src/develop/imageop.c

```diff
@@ -1698,10 +1717,47 @@ gboolean _iop_validate_params(dt_introspection_field_t *field, gpointer params,
   return all_ok;
 }
 
+static gboolean _iop_update_label(gpointer data)
+{
+  dt_iop_module_t *module = (dt_iop_module_t *)data;
+  const uint32_t mask_mode =
+    module->blend_params
+    ? module->blend_params->mask_mode
+    : 0;
+
+  char nl[256] = { 0 };
+  char *preset_name =
+    dt_presets_get_name(module->op,
+                        module->params, module->params_size,
+                        ((module->flags() & IOP_FLAGS_SUPPORTS_BLENDING)
+                         && mask_mode)
+                        ? module->blend_params
+                        : module->default_blendop_params,
+                        sizeof(dt_develop_blend_params_t));
+
+  if(preset_name)
+    snprintf(nl, sizeof(nl), "%s", preset_name);
+  else if(module->multi_priority != 0)
+    snprintf(nl, sizeof(nl), "%d", module->multi_priority);
+
+  g_free(preset_name);
+
+  g_strlcpy(module->multi_name, nl, sizeof(module->multi_name));
+
+  dt_iop_gui_update_header(module);
+
+  module->label_recompute_handle = 0;
+  return G_SOURCE_REMOVE;
+}
+
 void dt_iop_commit_params(dt_iop_module_t *module, dt_iop_params_t *params,
```

Comment:
If I understand correctly, your intention is to update when the pipe is calculated? Presumable you only care about it when loaded in darkroom? So before that name isn't necessarily set correctly? Does the changed name then get added to the history/saved? If not, does it get set again next time pipe is calculated?

Reply:
No, there is no relation with the pipe. The idea is to update the IOP label when parameters are updated, so for every user change and when the module is initially initialized. And yes the name is added into the saved history.